frontend/src/metabase/visualizations/types/click-actions.ts

```diff
@@ -80,11 +81,17 @@ type UrlClickActionBase = {
 
 export type UrlClickAction = ClickActionBase & UrlClickActionBase;
 
+export type CustomClickAction = ClickActionBase & {
+  type: "custom";
+  onClick?: (closePopover: () => void) => void;
```

Comment:
Fair enough, but I think we could leave it required until we add more props though. It would make sense from the reviewer's point of view too since we would make `onClick` optional because we add more props.